spec/gemfury/app_spec.rb

```diff
@@ -3,6 +3,16 @@
 describe Gemfury::Command::App do
   Endpoint = "www.gemfury.com"
   class MyApp < Gemfury::Command::App
+    desc "die!", "die!", :hide => true
+    define_method("die!") do |*a|
+      super
+    end
+
+    desc "__die!_without_any_instance__", "__die!_without_any_instance__", :hide => true
+    define_method("__die!_without_any_instance__") do |*a|
+      super
+    end
```

Comment:
I don't understand this. Did they change the underlying `die!` implementation? Will this affect calling `MyApp#die!` outside of specs?

Reply:
@rykov `rspec-mocks` (see [this line](https://github.com/rspec/rspec-mocks/blob/f45810aa3add135bd8235df600bbe4b446fc77d4/lib/rspec/mocks/any_instance/recorder.rb#L260) and [this one](https://github.com/rspec/rspec-mocks/blob/407a8aa70cb10ad2c79cc2b47576d2b1f5aa7b5e/lib/rspec/mocks/any_instance/recorder.rb#L236)) add these new methods at runtime to the MyApp class, as part of observing the method calls to `die!` which is triggered by this [line in Gemfury specs](https://github.com/gemfury/gemfury/blob/e2170bd84ba576431f2b46612717c22c23536b87/spec/gemfury/app_spec.rb#L338), which then Thor think it is a command.

I looked up Thor's code, and unless the `method_added` is invoked within a `no_commands` block, it will assume it is a command. Further we can not wrap it with `no_commands`, since the `observe!` method in `rspec-mocks` is called internally.

See the [method hook here](https://github.com/erikhuda/thor/blob/99330185faa6ca95e57b19a402dfe52b1eba8901/lib/thor/base.rb#L631), and the succeeding [create_command](https://github.com/erikhuda/thor/blob/fb625b223465692a9d8a88cc2a483e126f1a8978/lib/thor.rb#L415). and if no desc is defined, the else block is invoked and a warning shows up.

i looked for ways on how to pre-empt Thor about these methods, without disrupting `rspec-mocks` behaviour. and came up with this approach.

what do you think?